components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/util/WebSubHubAdapterUtil.java

```diff
@@ -0,0 +1,429 @@
+                                        eventContext.getTenantDomain())
+                                .inputParam(WebSubHubAdapterConstants.LogConstants.InputKeys.TOPIC, topic)
+                                .resultMessage("Event data published to WebSubHub.")
+                                .resultStatus(DiagnosticLog.ResultStatus.SUCCESS)
+                                .logDetailLevel(DiagnosticLog.LogDetailLevel.INTERNAL_SYSTEM);
+                        LoggerUtils.triggerDiagnosticLogEvent(diagnosticLogBuilder);
+                    }
+                    try {
+                        log.debug("Response data: " + EntityUtils.toString(response.getEntity()));
+                    } catch (IOException e) {
+                        log.debug("Error while reading WebSubHub event publisher response. ", e);
+                    }
+                } else {
+                    String errorResponseBody;
+                    if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                        DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                                WebSubHubAdapterConstants.LogConstants.WEB_SUB_HUB_ADAPTER,
+                                WebSubHubAdapterConstants.LogConstants.ActionIDs.PUBLISH_EVENT);
+                        diagnosticLogBuilder
+                                .inputParam(WebSubHubAdapterConstants.LogConstants.InputKeys.URL, url)
+                                .inputParam(WebSubHubAdapterConstants.LogConstants.InputKeys.TENANT_DOMAIN,
+                                        eventContext.getTenantDomain())
+                                .inputParam(WebSubHubAdapterConstants.LogConstants.InputKeys.TOPIC, topic)
+                                .resultMessage("Failed to publish event data to WebSubHub.")
+                                .resultStatus(DiagnosticLog.ResultStatus.FAILED)
+                                .logDetailLevel(DiagnosticLog.LogDetailLevel.INTERNAL_SYSTEM);
+                        LoggerUtils.triggerDiagnosticLogEvent(diagnosticLogBuilder);
+                    }
+                    try {
+                        errorResponseBody = EntityUtils.toString(response.getEntity());
+                        log.error("WebHubSub event publisher received " + responseCode + " code. Response data: " + errorResponseBody);
+                    } catch (IOException e) {
+                        log.error("Error while reading WebSubHub event publisher response. ", e);
+                    }
+                }
+            }
+
+            @Override
+            public void failed(final Exception ex) {
+
+                handleResponseCorrelationLog(request, requestStartTime, WebSubHubCorrelationLogUtils.RequestStatus.FAILED.getStatus(),
+                        ex.getMessage());
+                log.error("Publishing event data to WebSubHub failed. ", ex);
+            }
```

Comment:
well it's not a good practice to add a method for an exception

Reply:
The `failed` method provided is part of the `FutureCallback<HttpResponse>` implementation. It's designed to handle cases where the HTTP request fails due to an exception.